File: Front/Admin/src/core/dtos/create/createPizza.dto.ts
import type { PizzaSize } from "../../types/pizza.type";
import type { CreateIngredientDto } from "./createIngredient.dto";
import type { CreatePizzaPriceDto } from "./createPizzaPrice.dto";

export interface CreatePizzaDto {
  categoryName: string;
  name: string;
  description: string;
  rating: number;
  imageUrl: string;
  stock: boolean;
  top: boolean;
  size: PizzaSize;
  ingredients: CreateIngredientDto[];
  prices: CreatePizzaPriceDto[];
}

File: Back/Pizzas.Core/Dtos/Create/CreatePizzaDto.cs
using Pizzas.Core.Dtos.Read;
using Pizzas.Core.Enums;

namespace Pizzas.Core.Dtos.Create;

public class CreatePizzaDto
{
    public string CategoryName = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Rating { get; set; } = 5;
    public string ImageUrl { get; set; } = string.Empty;
    public bool Stock { get; set; }
    public bool Top { get; set; }
    public PizzaSize Size { get; set; }
    public ICollection<CreateIngredientDto> Ingredients { get; set; } = [];
    public ICollection<CreatePizzaPriceDto> Prices { get; set; } = [];
}